force-app/main/default/classes/JobApplicationStatusHelper.cls

```diff
@@ -0,0 +1,103 @@
+/**
+ * @description       : This helper Class will create Tasks based on the Status of the Job Application.
+ * @author            : Raghava
+ * @group             : Cloud Smiths
+ * @last modified on  : 07-20-2024
+ * @last modified by  : Raghava
+**/
+public with sharing class JobApplicationStatusHelper {
+
+    public static List<Task> tasksToBeCreated = new List<Task>();
+
+    public static void tasksBasedOnStatus(List<Job_Application__c> newJobApplications, Map<Id, Job_Application__c> oldJobApplicationsMap){
+
+        //Looping over all the Job Application records that are passed through After Update Trigger Context
+        for(Job_Application__c jobApp : newJobApplications){
+            //Check if this is an update operation and there is a change in the Status of the Job Application
+            if(oldJobApplicationsMap != null && jobApp.Status__c != oldJobApplicationsMap.get(jobApp.Id).Status__c){
+                handleStatusChange(jobApp.Status__c, jobApp);
+            }
+            //To create tasks when the Job Application is Inserted.
+            else {
+                handleStatusChange(jobApp.Status__c, jobApp);
+            }
+    }
+
+    if(!tasksToBeCreated.isEmpty()){
```

Comment:
Where is the tasksToBeCreated list populated? I see the list instantiated, but I may be missing where records are added to the list.

Reply:
In the Line 18 & 22, based on the Job Application records creation / updation, it calls the method handleStatusChange() method where based on the 'Status' of each of the record, createTask() method is again called to create the Tasks. In the createTasks() method, the records are added to the list and we are using the list to create records in bulk.

Reply:
Is the list being populated in your System debugs? The handleStatusChange method doesn't return anything, so I don't see where the list is being populated.

Reply:
@triden92

Yeah, this list hodls the record values irrespective if we return or not. We do not need to return anything from the handleStatusChange() and createTask() methods because the tasksToBeCreated list is a static variable. This means it is accessible to all the methods within the JobApplicationStatusHelper class without needing to pass it around or return it.

Since tasksToBeCreated is static, any method within the class can modify this list directly. The methods handleStatusChange() and createTask() add tasks to this list, and since the list is static, these tasks will persist and be available after the methods complete.